### PR TITLE
feat(app): update deck map modal in protocol setup

### PR DESCRIPTION
**Pull request by koji**
# Overview
update deck map in protocol setup
close RAUT-671

# Test Plan
- set up a protocol
- tap modules and deck
- tap map view

# Changelog
- add ModulesAndDeckMapViewModal and test
- update ProtocolSetupModulesAndDeck test

# Review requests

# Risk assessment
low

### CODE REVIEW

File: app/src/organisms/ProtocolSetupModulesAndDeck/__tests__/ModulesAndDeckMapViewModal.test.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+const mockSetShowDeckMapModal = jest.fn()
+const PROTOCOL_ANALYSIS = {
+  id: 'fake analysis',
+  status: 'completed',
+  labware: [],
+} as any
+const simpleV7Protocol = (_uncastedSimpleV7Protocol as unknown) as ProtocolAnalysisOutput
+
+describe('ModulesAndDeckMapViewModal', () => {
+  let props: React.ComponentProps<typeof ModulesAndDeckMapViewModal>
```
Comment (koji) [inline app/src/organisms/ProtocolSetupModulesAndDeck/__tests__/ModulesAndDeckMapViewModal.test.tsx:109]:
```
  ModulesAndDeckMapViewModal
    ✕ should render map view when ff is on (2 ms)

  ● ModulesAndDeckMapViewModal › should render map view when ff is on

    TypeError: Invalid attempt to destructure non-iterable instance.
    In order to be iterable, non-array objects must have a [Symbol.iterator]() method.

      61 |
      62 | describe('ModulesAndDeckMapViewModal', () => {
    > 63 |   let props: React.ComponentProps<typeof ModulesAndDeckMapViewModal>
         |                                     ^
      64 |
      65 |   beforeEach(() => {
      66 |     props = {

      at _nonIterableRest (app/src/organisms/ProtocolSetupModulesAndDeck/__tests__/ModulesAndDeckMapViewModal.test.tsx:63:37)
      at _slicedToArray (app/src/organisms/ProtocolSetupModulesAndDeck/__tests__/ModulesAndDeckMapViewModal.test.tsx:61:138)
      at Object.<anonymous> (app/src/organisms/ProtocolSetupModulesAndDeck/__tests__/ModulesAndDeckMapViewModal.test.tsx:98:5)
```

File: app/src/organisms/ProtocolSetupModulesAndDeck/__tests__/ModulesAndDeckMapViewModal.test.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    // when(mockBaseDeck)
+    //   .calledWith({
+    //     deckLayerBlocklist: getStandardDeckViewLayerBlockList(FLEX_ROBOT_TYPE),
+    //     deckConfig: EXTENDED_DECK_CONFIG_FIXTURE,
+    //     robotType: FLEX_ROBOT_TYPE,
+    //     labwareLocations: [],
+    //     moduleLocations: mockModuleLocations,
+    //   })
+    //   .mockReturnValue(<div>mock BaseDeck</div>)
+    mockBaseDeck.mockReturnValue(<div>mock BaseDeck</div>)
```
Comment (koji) [inline app/src/organisms/ProtocolSetupModulesAndDeck/__tests__/ModulesAndDeckMapViewModal.test.tsx]: still trying to fix this error.

```
  ● ModulesAndDeckMapViewModal › should render map view when ff is on

    BaseDeck(...): Nothing was returned from render. This usually means a return statement is missing. Or, to render nothing, return null.

      46 |   }
      47 |
    > 48 |   return [render(Component, { wrapper: ProviderWrapper }), store]
         |           ^
      49 | }
      50 |
```

**Comment by koji** [comment]
- [ ] fix test cases

File: app/src/organisms/ProtocolSetupModulesAndDeck/index.tsx [hunk 3]
```diff
@@ -61,23 +55,23 @@ import { ModuleWizardFlows } from '../ModuleWizardFlows'
...
-  'SLOT_10_EXPANSION',
-  'CALIBRATION_CUTOUTS',
-]
+// const OT3_STANDARD_DECK_VIEW_LAYER_BLOCK_LIST: string[] = [
+//   'DECK_BASE',
+//   'BARCODE_COVERS',
+//   'SLOT_SCREWS',
+//   'SLOT_10_EXPANSION',
+//   'CALIBRATION_CUTOUTS',
+// ]
```
Comment (brenthagen) [inline app/src/organisms/ProtocolSetupModulesAndDeck/index.tsx]: comment can be removed

File: app/src/organisms/ProtocolSetupModulesAndDeck/ModulesAndDeckMapViewModal.tsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+const OT3_STANDARD_DECK_VIEW_LAYER_BLOCK_LIST: string[] = [
+  'DECK_BASE',
+  'BARCODE_COVERS',
+  'SLOT_SCREWS',
+  'SLOT_10_EXPANSION',
+  'CALIBRATION_CUTOUTS',
+]
+
+// Note (kk:10/26/2023) once we are ready for removing ff, we will be able to update props
```
Comment (brenthagen) [inline app/src/organisms/ProtocolSetupModulesAndDeck/ModulesAndDeckMapViewModal.tsx:38]: we can leave the feature flag out of this component, the rest of the deck config feature flags will be removed soon

File: app/src/organisms/ProtocolSetupModulesAndDeck/ModulesAndDeckMapViewModal.tsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+}: ModulesAndDeckMapViewModalProps): JSX.Element | null {
+  const { t } = useTranslation('protocol_setup')
+  const enableDeckConfig = useFeatureFlag('enableDeckConfiguration')
+
+  const modalHeader: ModalHeaderBaseProps = {
+    title: t('map_view'),
+    hasExitIcon: true,
+  }
+
+  const storedProtocolAnalysis = useStoredProtocolAnalysis(runId)
```
Comment (brenthagen) [inline app/src/organisms/ProtocolSetupModulesAndDeck/ModulesAndDeckMapViewModal.tsx]: can remove `useStoredProtocolAnalysis`, we will use the robot analysis only on ODD

File: app/src/organisms/ProtocolSetupModulesAndDeck/ModulesAndDeckMapViewModal.tsx [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    >
+      {/* we will be able to remove this once we are ready for removing the ff */}
+      {enableDeckConfig ? (
+        <BaseDeck
+          deckConfig={deckConfig}
+          deckLayerBlocklist={getStandardDeckViewLayerBlockList(
+            FLEX_ROBOT_TYPE
+          )}
+          robotType={FLEX_ROBOT_TYPE}
+          labwareLocations={labwareLocations}
```
Comment (brenthagen) [inline app/src/organisms/ProtocolSetupModulesAndDeck/ModulesAndDeckMapViewModal.tsx]: want `labwareLocations={[]}` here, figma says modules only for this deck map

File: app/src/organisms/ProtocolSetupModulesAndDeck/__tests__/ModulesAndDeckMapViewModal.test.tsx [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+import { when, resetAllWhenMocks } from 'jest-when'
+
+import {
+  renderWithProviders,
+  BaseDeck,
+  RobotWorkSpace,
+  EXTENDED_DECK_CONFIG_FIXTURE,
+} from '@opentrons/components'
+// import { FLEX_ROBOT_TYPE } from '@opentrons/shared-data'
+// import { parseInitialLoadedLabwareByAdapter } from '@opentrons/api-client'
```
Comment (brenthagen) [inline app/src/organisms/ProtocolSetupModulesAndDeck/__tests__/ModulesAndDeckMapViewModal.test.tsx]: some comments to remove in this file

**Review by brenthagen — APPROVED** [review]
looks good, i can remove feature flag in RAUT-823

**Pull request merged**